back/app/controllers/web_api/v1/admin_publications_controller.rb

```diff
@@ -33,6 +33,24 @@ def reorder
     end
   end
 
+  def list_areas_of_projects
+    authorize :admin_publication, :list_areas_of_projects
+    
+    publications = policy_scope(AdminPublication).includes(:parent)
+    
+    project_publications = publications.where(publication_type: Project.name).where.not(publication_status: :draft)
+
+    children_of_non_draft_parents = project_publications.where.not(parent: { publication_status: :draft })
+    projects_without_parents      = project_publications.where(parent_id: nil)
+
+    visible_project_publications = children_of_non_draft_parents.or(projects_without_parents)
+
+    areas_ids = AreasProject.where(project_id: visible_project_publications.select(:publication_id)).select(:area_id)
+    areas = Area.where(id: areas_ids)
+
+    render json: { areas: areas }
```

Comment:
I feel like I should be using a serializer here. I am not sure how to do this, or exactly why. I have tried mimicking the serializer used in areas_controller index, but it seems to want pages specified (which are not needed here) and when I try specifying pages with appropriate query params, it still errors with `NoMethodError: undefined method `first_page?' for nil:NilClass`.  
  
I have tried doing a little research around json serializers in general, but haven't really got any closer to understanding our specific implementations.

In this case, the only significant difference I can see in the json produced is my version begins with `{"areas":[{...` and other, serialized, json produced by our codebase seems to begin with `{"data":[{...`. I have noticed that the format without `data` precludes using the `json_data` helper in specs, which is at least one drawback I can see.